src/intrinsics/ecma262/JSON.js

```diff
@@ -317,8 +317,17 @@ const JSONParse = buildExpressionTemplate(JSONParseStr);
 function InternalJSONClone(realm: Realm, val: Value): Value {
   if (val instanceof AbstractValue) {
     if (val instanceof AbstractObjectValue) {
-      let strVal = AbstractValue.createFromTemplate(realm, JSONStringify, StringValue, [val], JSONStringifyStr);
-      let obVal = AbstractValue.createFromTemplate(realm, JSONParse, ObjectValue, [strVal], JSONParseStr);
+      let strVal;
+      let obVal;
+
+      // If val is temporal, then this operation is also temporal. See #2327.
+      if (val.isTemporal()) {
```

Comment:
This logic seems the inverse of what you intend. This seems to call for an additional test case.

Reply:
It was inverted. On the other hand, it is not obvious to me that InternalJSONClone can create the particular conditions that this PR is trying to fix. So I have dropped this check for now, and will audit in a follow up request.